### PR TITLE
[Fixes #11579] Use autocomplete API for editing linked resources

### LINKED ISSUE #11579: Use autocomplete API for editing linked resources
At the moment the `Link To` widget inside the metadata editor (inherited from `ResourceBaseForm`) preloads all the resources that can be linked.
This is clearly unfeasible when we have a significant number of resources in the catalog. This problem is already visible in the stable demo when editing a document metadata the page takes a lot to load and the selection inside the widget is slow and sloppy.
This problem is exhacerbated now that we allow to link any type of resource.

We should convert that form field widget to `autocomplete.ModelSelect2` similar to [what we do for the owner](https://github.com/GeoNode/geonode/blob/master/geonode/base/forms.py#L408-L414)

### PR DESCRIPTION
**Pull request by etj**
#11579 Use autocomplete API for editing linked resources

## Checklist

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [x] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: black geonode && flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [x] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: geonode/base/views.py [hunk 1]
```diff
@@ -282,6 +282,28 @@ def get_queryset(self):
...
+        if self.forwarded and "exclude" in self.forwarded:
+            qs = qs.exclude(pk=self.forwarded["exclude"])
+
+        return get_visible_resources(
+            qs,
+            self.request.user if self.request else None,
+            admin_approval_required=settings.ADMIN_MODERATE_UPLOADS,
+            unpublished_not_visible=settings.RESOURCE_PUBLISHING,
+            private_groups_not_visibile=settings.GROUP_PRIVATE_RESOURCES,
+        )[:10]
```
Comment (mattiagiupponi) [inline geonode/base/views.py:301]: In this way always the first 10 rows are displayed so the pagination is not handled.
In another kind of this select the pagination is always available so the user can load more pages. Shouldn't this have the same behavior?

File: geonode/base/forms.py [hunk 2]
```diff
@@ -347,35 +349,45 @@ def _get_thesauro_title_label(item, lang):
...
-        return [[obj.id, f"{obj.title} ({obj.polymorphic_ctype.model})"] for obj in resources]
-
     def save_linked_resources(self, links_field="linked_resources"):
         # create and fetch desired links
         target_ids = []
-        for res_id in self.cleaned_data[links_field]:
-            linked, _ = LinkedResource.objects.get_or_create(source=self.instance, target_id=res_id, internal=False)
-            target_ids.append(res_id)
+        for res in self.cleaned_data[links_field]:
+            linked, _ = LinkedResource.objects.get_or_create(source=self.instance, target=res, internal=False)
```
Comment (mattiagiupponi) [inline geonode/base/forms.py:382]: SInce this `linked` and `_` are not used, is needed to define them?

**Review by mattiagiupponi — CHANGES_REQUESTED** [review]
Is it possible to add also a couple of tests for the other resources? I see them only for datasets

File: geonode/base/forms.py (hunk 2 above)
Comment (etj) [reply geonode/base/forms.py:382]: quite strange flake didnt spot it

File: geonode/base/views.py (hunk 1 above)
Comment (etj) [reply geonode/base/views.py:301]: Copied from ResourceBaseAutocomplete

**Review by giohappy — APPROVED** [review]

**Review by mattiagiupponi — APPROVED** [review]

**Pull request merged**